modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -107,9 +107,7 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
         obsId <- safeGetObsId(id)
         _ <- systems.odb.obsAbort(obsId, e.msg)
       } yield ()
-      case (EventSystem(Executed(id)), st) if st.sequences.get(id).map(
-        _.status === SequenceState.Idle
-      ).getOrElse(false) => for {
+      case (EventSystem(Executed(id)), st) if st.sequences.get(id).exists(_.status === SequenceState.Idle) => for {
```

Comment:
Oh, you are right, I forgot about how pattern matching works in Scala. I guess it could be done implementing `unapply` for `Event`s, but it's not worth doing it in this PR. I'll try to implement it at some point when dealing with the engine Events.